ui/src/components/Broker/BrokerOrders.tsx

```diff
@@ -136,58 +118,42 @@ type BrokerOrderCardProps = {
 
 
 const BrokerOrderCard: React.FC<BrokerOrderCardProps> = (props) => {
-    const [ loading, setLoading ] = useState(false);
+    const [ depositCid, setDepositCid ] = useState<string>('');
+    const broker = useParty();
+    const ledger = useLedger();
+
     const [ base, quote ] = unwrapDamlTuple(props.cdata.pair).map(t => t.label);
     const label = props.cdata.isBid ? `Buy ${base}/${quote}` : `Sell ${base}/${quote}`;
     const amount = props.cdata.isBid ? `+ ${props.cdata.qty} ${base}` : `- ${props.cdata.qty} ${base}`;
     const price = `${props.cdata.price} ${quote}`;
 
     const customer = props.cdata.brokerCustomer;
 
-    const [ depositCid, setDepositCid ] = useState<string>('');
-    const [ error, setError ] = useState<ErrorMessage>();
-    const ledger = useLedger();
-
-    const handleAcceptBrokerOrderFill = async (event: React.FormEvent) => {
-        event.preventDefault();
-        setLoading(true);
-        try {
-            const key = wrapDamlTuple([props.cdata.broker, props.cdata.brokerOrderId])
-            const args = {
-                depositCid
-            }
-            await ledger.exerciseByKey(BrokerOrder.BrokerOrder_Fill, key, args);
-            clearForm()
-        } catch (err) {
-            setError(parseError(err));
-        }
-        setLoading(false);
-    }
+    const options = props.deposits
+    .filter(deposit => deposit.contractData.account.owner === broker)
+    .map(deposit => ({
+        key: deposit.contractId,
+        value: deposit.contractId,
+        text: `${deposit.contractData.asset.quantity} ${deposit.contractData.asset.id.label} | ${deposit.contractData.account.id.label}`
+    }))
 
-    function clearForm() {
-        setDepositCid('')
+    const handleAcceptBrokerOrderFill = async () => {
+        const key = wrapDamlTuple([props.cdata.broker, props.cdata.brokerOrderId])
+        const args = { depositCid }
+        await ledger.exerciseByKey(BrokerOrder.BrokerOrder_Fill, key, args);
+        setDepositCid('');
```

Comment:
should we use `.then(setDepositCid(''))` here and in all other places where we clear the forms? (ie clear the form only on successful submission). The reasoning is that if the user inputed something that was wrong then hit submit, they would still have the option to correct it inline.

Reply:
Ah, this already happens as you describe. Everything in this function gets wrapped in a `try { } except { }`, so the execution of this expands out to 

```javascript
setLoading(true);
try {
        const key = wrapDamlTuple([props.cdata.broker, props.cdata.brokerOrderId])
        const args = { depositCid }
        await ledger.exerciseByKey(BrokerOrder.BrokerOrder_Fill, key, args);
        setDepositCid('');
} catch (err) {
        setError(parseError(err));
}
setLoading(false);
```

inside the FormErrorHandled.

So if an error is caught from the ledger API call, then it won't clear